src/main/java/build/buf/protovalidate/internal/celext/CustomOverload.java

```diff
@@ -284,6 +288,54 @@ private static Overload isIp() {
         null);
   }
 
+  /**
+   * Creates a custom function overload for the "isIpPrefix" operation.
+   *
+   * @return The {@link Overload} instance for the "isIpPrefix" operation.
+   */
+  private static Overload isIpPrefix() {
+    return Overload.overload(
+        OVERLOAD_IS_IP_PREFIX,
+        null,
+        value -> {
+          if (value.type().typeEnum() != TypeEnum.String
+              && value.type().typeEnum() != TypeEnum.Bool) {
+            return Err.noSuchOverload(value, OVERLOAD_IS_IP_PREFIX, null);
+          }
+          String prefix = (String) value.value();
+          if (prefix.isEmpty()) {
+            return BoolT.False;
+          }
+          return Types.boolOf(validateIPPrefix(prefix, 0L, false));
+        },
+        (lhs, rhs) -> {
+          if (lhs.type().typeEnum() != TypeEnum.String
+              || (rhs.type().typeEnum() != TypeEnum.Int
+                  && rhs.type().typeEnum() != TypeEnum.Bool)) {
+            return Err.noSuchOverload(lhs, OVERLOAD_IS_IP_PREFIX, rhs);
+          }
+          String prefix = (String) lhs.value();
+          if (prefix.isEmpty()) {
+            return BoolT.False;
+          }
+          if (rhs.type().typeEnum() == TypeEnum.Int) {
+            return Types.boolOf(validateIPPrefix(prefix, rhs.intValue(), false));
+          }
+          return Types.boolOf(validateIPPrefix(prefix, 0L, rhs.booleanValue()));
+        },
+        (values) -> {
+          if (values.length != 3) {
```

Comment:
I think we still want to check the types of all fields here as before, just wanted the length check as well first.

Reply:
I reverted checking types.